tests/test_virtual_chassis.py

```diff
@@ -24,3 +26,96 @@ def test_connectivity(self, vct):
+
+    def test_voq_switch(self, vct):
+        """ Test VOQ switch objects configuration """
+        dvss = vct.dvss
+        for name in dvss.keys():
+            if not name.startswith("supervisor"):
+                print("VOQ Switch test for {}".format(name))
+                dvs = dvss[name]
+                #Get the config info
+                config_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+                metatbl = swsscommon.Table(config_db, "DEVICE_METADATA")
+
+                status, cfg_switch_id = metatbl.hget("localhost", "switch_id")
+                assert status, "Got error in getting switch_id from CONFIG_DB DEVICE_METADATA"
+
+                status, cfg_max_cores = metatbl.hget("localhost", "max_cores")
+                assert status, "Got error in getting max_cores from CONFIG_DB DEVICE_METADATA"
+                
+                cfgsptbl = swsscommon.Table(config_db, "SYSTEM_PORT")
+                cfgspkeys = cfgsptbl.getKeys()
+                sp_count = len(cfgspkeys)
+
+                asic_db = swsscommon.DBConnector(swsscommon.ASIC_DB, dvs.redis_sock, 0)
+                tbl = swsscommon.Table(asic_db, "ASIC_STATE:SAI_OBJECT_TYPE_SWITCH")
+                keys = list(tbl.getKeys())
+                switch_oid_key = keys[0]
+
+                status, value = tbl.hget(switch_oid_key, "SAI_SWITCH_ATTR_TYPE")
+                assert status, "Got error while getting switch type"
+                assert value == "SAI_SWITCH_TYPE_VOQ", "Switch type is not VOQ"
+                
+                status, value = tbl.hget(switch_oid_key, "SAI_SWITCH_ATTR_SWITCH_ID")
+                assert status, "Got error while getting switch id"
+                assert value == cfg_switch_id, "VOQ switch id is invalid"
+                
+                status, value = tbl.hget(switch_oid_key, "SAI_SWITCH_ATTR_MAX_SYSTEM_CORES")
+                assert status, "Got error while getting max system cores"
+                assert value == cfg_max_cores, "Max system cores is invalid"
+                
+                status, value = tbl.hget(switch_oid_key, "SAI_SWITCH_ATTR_SYSTEM_PORT_CONFIG_LIST")
+                assert status, "Got error while getting system port config list"
+                #Convert the spcfg string to dictionary
+                spcfg = ast.literal_eval(value)
+                assert spcfg['count'] == sp_count, "Number of systems ports configured is invalid"
+    
+    def test_chassis_app_db_sync(self, vct):
+        """ Test chassis app db syncing """
+        dvss = vct.dvss
+        for name in dvss.keys():
+            if name.startswith("supervisor"):
+                dvs = dvss[name]
+                chassis_app_db = swsscommon.DBConnector(swsscommon.CHASSIS_APP_DB, dvs.redis_chassis_sock, 0)
+                tbl = swsscommon.Table(chassis_app_db, "SYSTEM_INTERFACE")
+                keys = list(tbl.getKeys())
+                assert len(keys), "No chassis app db syncing is done"
```

Comment:
Should we validate that the number of system intfs is the same as system ports?

Reply:
Objective of this test is to verify successful syncing from line cards to supervisor card.  I just picked up one table to verity the syncing. It could be SYSTEM_NEIGH too. The next test "test_chassis_system_interface" does the actual verification of a sample system interface correctness.

We can validate the number of system interfaces, if the topology is finalized and we know exactly how many interfaces are expected to be synced.